src/Compilers/CSharp/Portable/Symbols/EventSymbol.cs

```diff
@@ -168,8 +168,7 @@ internal bool HidesBaseEventsByName
 
         internal EventSymbol GetLeastOverriddenEvent(NamedTypeSymbol? accessingTypeOpt)
         {
-            var accessingType = ((object?)accessingTypeOpt == null ? this.ContainingType : accessingTypeOpt).OriginalDefinition;
-
+            accessingTypeOpt = accessingTypeOpt?.OriginalDefinition;
```

Comment:
Can you elaborate on how multiple compilations played into this bug? Looking through the code I can't see why it too multiple compilations to produce this bug instead of just reproducing all in a single source file.

Reply:
With the repro (with a single assembly or multiple assemblies), overload resolution calls `PropertySymbol.GetLeastOverriddenProperty()` several times, including once for `Derived<B>.this[A] { get; }`, with `accessingTypeOpt == null`. Before this change, `GetLeastOverriddenProperty()` would check that the overridden property, `Base<A, B>.this[A] { get; }` was accessible within `accessingType` which was set to `Derived<TValue>`. In the single assembly case, all types are accessible. In the multi-assembly case, `B` is not accessible from the assembly containing `Derived<TValue>`. With this change, the access check is skipped here when `accessingTypeOpt == null`.


---
In reply to: [465822765](https://github.com/dotnet/roslyn/pull/46556#discussion_r465822765) [](ancestors = 465822765)